purchase_ext_sst/models/purchase_order.py

```diff
@@ -84,113 +95,96 @@ def open_record(self):
-                                                          mobile)
-                    if not partner_e:
-                        order.partner_id = self.create_partner(phone, mobile)
-                    self.update_partner(order.partner_id, phone, mobile,
-                                        tent_name)
-            else:
-                if val_phone or val_mobile or val_tent_name:
-                    partner_e = self.get_partner_from_phone(phone,
-                                                            mobile)
-                    if partner_e and order.partner_id != partner_e:
-                        self.update_partner(partner_e, phone, mobile,
-                                            tent_name)
-                        self.set_particulars_from_partner(partner_e, phone,
-                                                          mobile)
-                    else:
-                        self.update_partner(order.partner_id, phone, mobile,
-                                            tent_name)
-
-            if 'order_line' in vals:
-                for line in order.order_line:
-                    product = line.product_id
-                    if product.shop_id != order.shop_id:
-                        product.shop_id = order.shop_id
-                    if product.purchased_by_id != order.purchased_by_id:
-                        product.purchased_by_id = order.purchased_by_id
-        return res
-
-    @api.model
-    def create(self, vals):
-        val_phone = 'phone' in vals and vals['phone'] or False
-        val_mobile = 'mobile' in vals and vals['mobile'] or False
-        val_tent_name = 'tentative_name' in vals and vals['tentative_name'] \
+            val_phone_search = 'phone_search' in vals and vals[
+                'phone_search'] or False
+            val_phone = 'phone_update' in vals and vals['phone_update'] or \
+                        val_phone_search or False
+            val_mobile = 'mobile_update' in vals and vals['mobile_update'] \
                          or False
-        if self.is_default_partner(vals['partner_id']):
-            if val_phone or val_mobile:
-                partner_e = self.get_partner_from_phone(val_phone, val_mobile)
-                if partner_e and vals['partner_id'] != partner_e.id:
-                    vals['partner_id'] = partner_e.id
-                    if not vals['phone']:
-                        vals['phone'] = partner_e.phone
-                    if not vals['mobile']:
-                        vals['mobile'] = partner_e.mobile
-                if not partner_e:
+            val_tent_name = 'tentative_name' in vals and vals[
+                'tentative_name'] or order.tentative_name
+            vals['partner_id'] = 'partner_id' in vals and vals['partner_id']\
+                                 or order.partner_id.id
+            partner = self.get_partner_from_phone(val_phone, val_mobile)
```

Comment:
We should not need this if partner is proposed from `onchange_phone_search()` method.